File: boj/b_1149.js
function solution(n, array) {
  const table = Array.from({ length: n }, () => Array(3).fill(0));
  table[0] = [...array[0]];
  for (let i = 1; i < n; i += 1) {
    table[i][0] = array[i][0] + Math.min(table[i - 1][1], table[i - 1][2]);
    table[i][1] = array[i][1] + Math.min(table[i - 1][0], table[i - 1][2]);
    table[i][2] = array[i][2] + Math.min(table[i - 1][0], table[i - 1][1]);
  }
  return Math.min(...table[n - 1]);
}
console.log(
  solution(3, [
    [1, 100, 100],
    [100, 100, 100],
    [1, 100, 100],
  ])
);

File: boj/b_1149.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]
table = [arr[0]] + [[0] * 3 for _ in range(n - 1)]

for i in range(1, n):
    table[i][0] = arr[i][0] + min(table[i - 1][1], table[i - 1][2])
    table[i][1] = arr[i][1] + min(table[i - 1][0], table[i - 1][2])
    table[i][2] = arr[i][2] + min(table[i - 1][0], table[i - 1][1])
print(min(table[-1]))